File: app/src/main/java/io/shelfy/architecture/presentation/details/viewmodel/formatters/MovieDescriptionFormatter.java
package io.shelfy.architecture.presentation.details.viewmodel.formatters;

import android.graphics.Color;
import android.text.Spannable;
import android.text.SpannableStringBuilder;
import android.text.style.BackgroundColorSpan;

import io.shelfy.architecture.domain.entity.Movie;

class MovieDescriptionFormatter {

    public Spannable format(Movie movie) {
        String description = movie.getDescription();
        final SpannableStringBuilder formattedDescription = new SpannableStringBuilder(description);
        formattedDescription.setSpan(
                new BackgroundColorSpan(Color.CYAN),
                0,
                description.length(),
                Spannable.SPAN_EXCLUSIVE_EXCLUSIVE
        );
        return formattedDescription;
    }
}

File: app/src/main/java/io/shelfy/architecture/presentation/view/formatters/MovieDescriptionFormatter.kt
package io.shelfy.architecture.presentation.view.formatters

import android.graphics.Color
import android.text.Spannable
import android.text.SpannableStringBuilder
import android.text.style.BackgroundColorSpan
import io.shelfy.architecture.domain.entity.Movie

class MovieDescriptionFormatter {

    fun format(movieJson: Movie): Spannable {
        val description = movieJson.description
        return SpannableStringBuilder(description).apply {
            setSpan(
                BackgroundColorSpan(Color.CYAN),
                0,
                description.length,
                Spannable.SPAN_EXCLUSIVE_EXCLUSIVE
            )
        }
    }
}
